compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1572,30 +1572,28 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             return;
         }
 
+        let mut values = None;
+
         self.probe(|_| {
+            let ocx = ObligationCtxt::new_in_snapshot(self);
             let mut err = error.err;
-            let mut values = None;
 
             // try to find the mismatched types to report the error with.
             //
             // this can fail if the problem was higher-ranked, in which
             // cause I have no idea for a good error message.
             let bound_predicate = predicate.kind();
             if let ty::PredicateKind::Projection(data) = bound_predicate.skip_binder() {
-                let mut selcx = SelectionContext::new(self);
                 let data = self.replace_bound_vars_with_fresh_vars(
                     obligation.cause.span,
                     infer::LateBoundRegionConversionTime::HigherRankedType,
                     bound_predicate.rebind(data),
                 );
-                let mut obligations = vec![];
-                let normalized_ty = super::normalize_projection_type(
-                    &mut selcx,
-                    obligation.param_env,
-                    data.projection_ty,
+                let normalized_ty = ocx.normalize(
```

Comment:
After #104905, it's probably better for this to just use `self.at(..).normalize(..)` instead of making a whole `ObligationCtxt`.